build error on debian sid ( format not a string literal and no format arguments )

hi dargargo,

when i compile elektroid 2.2 with the current debian sid version i run into this build error:

./configure --build=x86_64-linux-gnu --prefix=/usr --includedir=\${prefix}/include --mandir=\${prefix}/share/man --infodir=\${prefix}/share/info --sysconfdir=/etc --localstatedir=/var --disable-option-checking --disable-silent-rules --libdir=\${prefix}/lib/x86_64-linux-gnu --runstatedir=/run --disable-maintainer-mode --disable-dependency-tracking
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking how to print strings... printf
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking for a sed that does not truncate output... /bin/sed
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for fgrep... /bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking for gawk... no
checking for mawk... mawk
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /bin/dd
checking how to truncate binary pipes... /bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking for library containing sqrt... -lm
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /bin/mkdir -p
checking whether make sets $(MAKE)... yes
checking whether make supports the include directive... yes (GNU style)
checking whether make supports nested variables... yes
checking dependency style of gcc... none
checking for gcc... (cached) gcc
checking whether the compiler supports GNU C... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to enable C11 features... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for alsa >= 1.1.3... yes
checking for zlib >= 1.1.8... yes
checking for libzip >= 1.1.2... yes
checking for gtk+-3.0... yes
checking for sndfile >= 1.0.2... yes
checking for samplerate >= 0.1.8... yes
checking for libpulse >= 5.0... yes
checking whether NLS is requested... yes
checking for msgfmt... /usr/bin/msgfmt
checking for gmsgfmt... /usr/bin/msgfmt
checking for xgettext... /usr/bin/xgettext
checking for msgmerge... /usr/bin/msgmerge
checking for ld used by gcc... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking for shared library run path origin... done
checking how to run the C preprocessor... gcc -E
checking for CFPreferencesCopyAppValue... no
checking for CFLocaleCopyCurrent... no
checking for GNU gettext in libc... yes
checking whether to use NLS... yes
checking where the gettext function comes from... libc
checking for unistd.h... (cached) yes
checking for limits.h... yes
checking for GNU libc compatible malloc... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating res/Makefile
config.status: creating test/Makefile
config.status: creating po/Makefile.in
config.status: creating po/Makefile
config.status: creating man/Makefile
config.status: creating config.h
config.status: executing libtool commands
config.status: executing depfiles commands
config.status: executing po-directories commands
config.status: creating po/POTFILES
config.status: creating po/Makefile
   dh_auto_build
	make -j4
make[1]: Entering directory '/build/elektroid-2.2'
make  all-recursive
make[2]: Entering directory '/build/elektroid-2.2'
Making all in src
make[3]: Entering directory '/build/elektroid-2.2/src'
gcc -DHAVE_CONFIG_H -I. -I..  -Wall -O3 -DDATADIR='"/usr/share/elektroid"' -DLOCALEDIR='"/usr/share/locale"' -Wdate-time -D_FORTIFY_SOURCE=2 -I../src `/usr/bin/pkg-config --cflags alsa glib-2.0 zlib json-glib-1.0 libzip gtk+-3.0 libpulse libpulse-mainloop-glib` -I/usr/include/opus -I/usr/include/x86_64-linux-gnu   -D_GNU_SOURCE -g -O2 -ffile-prefix-map=/build/elektroid-2.2=. -fstack-protector-strong -Wformat -Werror=format-security -c -o elektroid-local.o `test -f 'local.c' || echo './'`local.c
gcc -DHAVE_CONFIG_H -I. -I..  -Wall -O3 -DDATADIR='"/usr/share/elektroid"' -DLOCALEDIR='"/usr/share/locale"' -Wdate-time -D_FORTIFY_SOURCE=2 -I../src `/usr/bin/pkg-config --cflags alsa glib-2.0 zlib json-glib-1.0 libzip gtk+-3.0 libpulse libpulse-mainloop-glib` -I/usr/include/opus -I/usr/include/x86_64-linux-gnu   -D_GNU_SOURCE -g -O2 -ffile-prefix-map=/build/elektroid-2.2=. -fstack-protector-strong -Wformat -Werror=format-security -c -o elektroid-connector.o `test -f 'connector.c' || echo './'`connector.c
gcc -DHAVE_CONFIG_H -I. -I..  -Wall -O3 -DDATADIR='"/usr/share/elektroid"' -DLOCALEDIR='"/usr/share/locale"' -Wdate-time -D_FORTIFY_SOURCE=2 -I../src `/usr/bin/pkg-config --cflags alsa glib-2.0 zlib json-glib-1.0 libzip gtk+-3.0 libpulse libpulse-mainloop-glib` -I/usr/include/opus -I/usr/include/x86_64-linux-gnu   -D_GNU_SOURCE -g -O2 -ffile-prefix-map=/build/elektroid-2.2=. -fstack-protector-strong -Wformat -Werror=format-security -c -o elektroid-sample.o `test -f 'sample.c' || echo './'`sample.c
gcc -DHAVE_CONFIG_H -I. -I..  -Wall -O3 -DDATADIR='"/usr/share/elektroid"' -DLOCALEDIR='"/usr/share/locale"' -Wdate-time -D_FORTIFY_SOURCE=2 -I../src `/usr/bin/pkg-config --cflags alsa glib-2.0 zlib json-glib-1.0 libzip gtk+-3.0 libpulse libpulse-mainloop-glib` -I/usr/include/opus -I/usr/include/x86_64-linux-gnu   -D_GNU_SOURCE -g -O2 -ffile-prefix-map=/build/elektroid-2.2=. -fstack-protector-strong -Wformat -Werror=format-security -c -o elektroid-utils.o `test -f 'utils.c' || echo './'`utils.c
gcc -DHAVE_CONFIG_H -I. -I..  -Wall -O3 -DDATADIR='"/usr/share/elektroid"' -DLOCALEDIR='"/usr/share/locale"' -Wdate-time -D_FORTIFY_SOURCE=2 -I../src `/usr/bin/pkg-config --cflags alsa glib-2.0 zlib json-glib-1.0 libzip gtk+-3.0 libpulse libpulse-mainloop-glib` -I/usr/include/opus -I/usr/include/x86_64-linux-gnu   -D_GNU_SOURCE -g -O2 -ffile-prefix-map=/build/elektroid-2.2=. -fstack-protector-strong -Wformat -Werror=format-security -c -o elektroid-backend.o `test -f 'backend.c' || echo './'`backend.c
gcc -DHAVE_CONFIG_H -I. -I..  -Wall -O3 -DDATADIR='"/usr/share/elektroid"' -DLOCALEDIR='"/usr/share/locale"' -Wdate-time -D_FORTIFY_SOURCE=2 -I../src `/usr/bin/pkg-config --cflags alsa glib-2.0 zlib json-glib-1.0 libzip gtk+-3.0 libpulse libpulse-mainloop-glib` -I/usr/include/opus -I/usr/include/x86_64-linux-gnu   -D_GNU_SOURCE -g -O2 -ffile-prefix-map=/build/elektroid-2.2=. -fstack-protector-strong -Wformat -Werror=format-security -c -o elektroid-audio.o `test -f 'audio.c' || echo './'`audio.c
gcc -DHAVE_CONFIG_H -I. -I..  -Wall -O3 -DDATADIR='"/usr/share/elektroid"' -DLOCALEDIR='"/usr/share/locale"' -Wdate-time -D_FORTIFY_SOURCE=2 -I../src `/usr/bin/pkg-config --cflags alsa glib-2.0 zlib json-glib-1.0 libzip gtk+-3.0 libpulse libpulse-mainloop-glib` -I/usr/include/opus -I/usr/include/x86_64-linux-gnu   -D_GNU_SOURCE -g -O2 -ffile-prefix-map=/build/elektroid-2.2=. -fstack-protector-strong -Wformat -Werror=format-security -c -o elektroid-browser.o `test -f 'browser.c' || echo './'`browser.c
gcc -DHAVE_CONFIG_H -I. -I..  -Wall -O3 -DDATADIR='"/usr/share/elektroid"' -DLOCALEDIR='"/usr/share/locale"' -Wdate-time -D_FORTIFY_SOURCE=2 -I../src `/usr/bin/pkg-config --cflags alsa glib-2.0 zlib json-glib-1.0 libzip gtk+-3.0 libpulse libpulse-mainloop-glib` -I/usr/include/opus -I/usr/include/x86_64-linux-gnu   -D_GNU_SOURCE -g -O2 -ffile-prefix-map=/build/elektroid-2.2=. -fstack-protector-strong -Wformat -Werror=format-security -c -o elektroid-notifier.o `test -f 'notifier.c' || echo './'`notifier.c
backend.c: In function 'backend_midi_handshake':
backend.c:205:28: error: format not a string literal and no format arguments [-Werror=format-security]
  205 |                     backend->device_name);
      |                     ~~~~~~~^~~~~~~~~~~~~
gcc -DHAVE_CONFIG_H -I. -I..  -Wall -O3 -DDATADIR='"/usr/share/elektroid"' -DLOCALEDIR='"/usr/share/locale"' -Wdate-time -D_FORTIFY_SOURCE=2 -I../src `/usr/bin/pkg-config --cflags alsa glib-2.0 zlib json-glib-1.0 libzip gtk+-3.0 libpulse libpulse-mainloop-glib` -I/usr/include/opus -I/usr/include/x86_64-linux-gnu   -D_GNU_SOURCE -g -O2 -ffile-prefix-map=/build/elektroid-2.2=. -fstack-protector-strong -Wformat -Werror=format-security -c -o elektroid-preferences.o `test -f 'preferences.c' || echo './'`preferences.c
cc1: some warnings being treated as errors
make[3]: *** [Makefile:676: elektroid-backend.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/build/elektroid-2.2/src'
make[2]: *** [Makefile:438: all-recursive] Error 1
make[2]: Leaving directory '/build/elektroid-2.2'
make[1]: *** [Makefile:370: all] Error 2
make[1]: Leaving directory '/build/elektroid-2.2'

--- a/src/backend.c
+++ b/src/backend.c
@@ -201,7 +201,7 @@ backend_midi_handshake (struct backend *backend)
 		    backend->midi_info.version[1],
 		    backend->midi_info.version[2],
 		    backend->midi_info.version[3]);
-	  snprintf (backend->device_desc.name, LABEL_MAX,
+	  snprintf (backend->device_desc.name, LABEL_MAX, "%s",
 		    backend->device_name);
 	  debug_print (1, "Detected device: %s\n", backend->device_name);
 	}

--- a/src/connectors/efactor.c
+++ b/src/connectors/efactor.c
@@ -532,7 +532,7 @@ efactor_handshake (struct backend *backend)
 		    backend->midi_info.version[1],
 		    backend->midi_info.version[2],
 		    backend->midi_info.version[3]);
-	  snprintf (backend->device_desc.name, LABEL_MAX,
+	  snprintf (backend->device_desc.name, LABEL_MAX, "%s",
 		    backend->device_name);
 	  debug_print (1, "XML version:\n%s\n", &rx_msg->data[14]);
 	}

